packages/sdk/src/sign.ts

```diff
@@ -217,6 +231,31 @@ export const unpackEnvelope = async (
     return makeParsedEvent(event, envelope.hash, envelope.signature)
 }
 
+export const unpackSnapshot = async (
+    headerEvent: StreamEvent,
+    snapshot: Envelope,
+    opts: UnpackEnvelopeOpts | undefined,
+): Promise<ParsedSnapshot> => {
+    check(hasElements(snapshot.event), 'Snapshot base is not set', Err.BAD_EVENT)
+    check(hasElements(snapshot.hash), 'Snapshot hash is not set', Err.BAD_EVENT)
+    check(hasElements(snapshot.signature), 'Snapshot signature is not set', Err.BAD_EVENT)
+
+    const sn = fromBinary(SnapshotSchema, snapshot.event)
+    let hash = snapshot.hash
+
+    if (opts?.disableHashValidation !== true) {
+        hash = riverSnapshotHash(snapshot.event)
+        check(bin_equal(hash, snapshot.hash), 'Snapshot id is not valid', Err.BAD_EVENT_ID)
+    }
+
+    const doCheckEventSignature = opts?.disableSignatureValidation !== true
+    if (doCheckEventSignature) {
+        checkEventSignature(headerEvent, hash, snapshot.signature)
```

Comment:
would love to either see a comment here that the headerEvent contains the createrAddress of the snapshot OR, call this like so 
```
checkEventSignature({
        creatorAddress: headerEvent.creatorAddress
        delegateSig: headerEvent.delegateSig
        delegateExpiryEpochMs: headerEvent.delegateExpiryEpochMs
    },
    hash,
    snapshot.signature
}
```
for clarity